### DISCUSSION TITLE
meshToSignedDistanceField values outside those occupied by the mesh?

### DISCUSSION THREAD

**Discussion by FrankEscobar** [AcademySoftwareFoundation/openvdb#1549] [Q&A] [answered]
I'm trying to make something experimental but the value retrieved is always the background value instead of the distance to the mesh when I'm not on a voxel that intersects with the geometry.

Actually I'm not interested on the value itself but on the sign (is the voxel inside or outside the geometry?)

Is there any chance to force that? Regards.

**Comment by Idclip** [comment, marked as answer]
> I'm trying to make something experimental but the value retrieved is always the background value instead of the distance to the mesh when I'm not on a voxel that intersects with the geometry.

This is the premise and design of narrow band SDFs in OpenVDB. When a SDF is constructed you choose the size of the external/internal bands which control how far away from the surface you want to store valid distance values. Voxels further away (in index space) than this band size will receive constant background values. You can increase the band sizes as desired.

> Actually I'm not interested on the value itself but on the sign (is the voxel inside or outside the geometry?)

SDFs created through OpenVDB methods will flood fill the _outside_ of the level set with **inactive positive** background values. The _inside_ of the level set will be filled with **inactive negative** background sizes. You are able to check the sign of any value in a valid SDF and know immediately from the sign whether you are outside (+) or inside (-)

**Reply by FrankEscobar** [reply]
Thank you!
